src/migrations/migration1.ts

```diff
@@ -0,0 +1,55 @@
+import { Migration } from './interface';
+import { getDataForPath, clearStorage, setDataToPath, serializeMap } from '../common/HelperFunctions';
+/**
+ * Converts obj in each path (query, search and execute) to
+ * new data structure into map
+ */
+class Migration1 implements Migration {
+  private allData: Object;
+  private extensionState: Object;
+  private statesKey = 'extension_states';
+
+  async shouldUpgrade() {
+    this.allData = await getDataForPath(null); // 'null' gets all the data for all paths
+    if (this.allData.hasOwnProperty(this.statesKey)) {
+      this.extensionState = this.allData[this.statesKey];
```

Comment:
`extension_states` what is exactly stored in it?

Reply:
Open or close state of each page in extension. E.g.,
```
{
  "query": true,
  "search": false
}

True => Extension was open when last updated. 
False => Extension was closed when last updated